File: Latihan5/C++/soal1.cpp
#include <iostream>
#include <string>
#include <iomanip>

using namespace std;

// Fungsi untuk mengkonversi nilai huruf ke bobot nilai
double konversiNilaiKeBobot(string nilai) {
    if (nilai == "A") return 4.0;
    if (nilai == "A-") return 3.7;
    if (nilai == "B+") return 3.3;
    if (nilai == "B") return 3.0;
    if (nilai == "B-") return 2.7;
    if (nilai == "C+") return 2.3;
    if (nilai == "C") return 2.0;
    if (nilai == "D") return 1.0;
    if (nilai == "E") return 0.0;
    return -1.0;  // Mengembalikan -1 jika nilai tidak valid
}

int main() {
    int jumlahMataKuliah;
    double totalSKS = 0.0;
    double totalNilai = 0.0;

    cout << "Masukkan jumlah mata kuliah: ";
    cin >> jumlahMataKuliah;

    for (int i = 1; i <= jumlahMataKuliah; i++) {
        string namaMatkul;
        int sks;
        string nilai;

        cout << "Nama Mata Kuliah ke-" << i << ": ";
        cin.ignore();  // Menghapus karakter newline sebelum membaca string
        getline(cin, namaMatkul);

        cout << "Jumlah SKS Mata Kuliah " << namaMatkul << ": ";
        cin >> sks;

        cout << "Nilai Mata Kuliah " << namaMatkul << " (A, A-, B+, B, B-, C+, C, D, atau E): ";
        cin >> nilai;

        double bobotNilai = konversiNilaiKeBobot(nilai);

        if (bobotNilai == -1.0) {
            cout << "Nilai tidak valid. Silakan masukkan nilai yang benar." << endl;
            return 1;  // Keluar program dengan kode kesalahan
        }

        totalSKS += sks;
        totalNilai += bobotNilai * sks;
    }

    double ipk = totalNilai / totalSKS;

    cout << fixed << setprecision(2);
    cout << "IPK Anda adalah: " << ipk << endl;

    return 0;
}

File: Latihan5/Python/soal1.py
def hitung_ipk():
    totalSks = 0
    totalNilai = 0

    jumlahMatkul = int(input("Masukkan jumlah mata kuliah: "))

    for i in range(jumlahMatkul):
        namaMatkul = input(f"Nama Mata Kuliah ke-{i + 1}: ")
        sks = int(input(f"Jumlah SKS Mata Kuliah {namaMatkul}: "))
        nilai = input(f"Nilai Mata Kuliah {namaMatkul} (A, A-, B+, B, B-, C+, C, D, atau E): ")

        if nilai == "A":
            bobot_nilai = 4.0
        elif nilai == "A-":
            bobot_nilai = 3.7
        elif nilai == "B+":
            bobot_nilai = 3.3
        elif nilai == "B":
            bobot_nilai = 3.0
        elif nilai == "B-":
            bobot_nilai = 2.7
        elif nilai == "C+":
            bobot_nilai = 2.3
        elif nilai == "C":
            bobot_nilai = 2.0
        elif nilai == "D":
            bobot_nilai = 1.0
        elif nilai == "E":
            bobot_nilai = 0.0
        else:
            print("Nilai tidak valid. Silakan masukkan nilai yang benar.")
            return

        totalSks += sks
        totalNilai += (bobot_nilai * sks)

    ipk = totalNilai / totalSks
    print(f"IPK Anda adalah: {ipk:.2f}")

if __name__ == "__main__":
    hitung_ipk()
